packages/solana-wallet-injection/src/wallet.ts

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable no-console */
+import { Liquality, LiqualityEvent } from '@liquality/solana-wallet-standard/src/window';
+import {
+  PublicKey,
+  Transaction,
+  VersionedTransaction,
+  SendOptions,
+  Connection,
+  Message,
+  Keypair,
+} from '@solana/web3.js';
+import { SolanaParser, ParsedInstruction } from '@debridge-finance/solana-transaction-parser';
+import { Idl } from '@project-serum/anchor';
+import { COMMON_REQUEST_MAP } from './types';
+import { BigNumber } from '@chainify/types';
+import base58 from 'bs58';
+
+/**
+ * @dev users are paying a fixed fee of 5000 lamports per transaction/signature
+ */
+const DEFAULT_FEE = '0.000005';
+
+export class LiqualitySolanaWallet implements Liquality {
+  public publicKey: PublicKey | null;
+  public isConnected: boolean;
+  private window: Window;
+  private _connection: Connection;
+
+  public get connection(): Connection {
+    return this._connection;
+  }
+  public set connection(value: Connection) {
+    this._connection = value;
+  }
+  private solanaParser = new SolanaParser([]);
+
+  constructor(window: Window) {
+    this.window = window;
+    // TODO: take rpc url from cryptoassets
+    this.connection = new Connection(
+      'https://red-sleek-rain.solana-mainnet.discover.quiknode.pro/fc112deb1e0228d09d0d8c12b8de5a601d251d80'
+    );
+  }
+
+  public async connect(
+    _options?: { onlyIfTrusted?: boolean | undefined } | undefined
+  ): Promise<{ publicKey: PublicKey }> {
+    const { accepted } = await this.enable();
+
+    if (!accepted) {
+      throw new Error('User rejected');
+    }
+
+    const addresses = await this.callMethod(COMMON_REQUEST_MAP.wallet_getAddresses);
+    const [{ publicKey }] = addresses;
+    this.publicKey = new PublicKey(publicKey);
+    this.isConnected = true;
+
+    console.debug(`Public Key`, this.publicKey);
+    console.debug(`Public Key Decoded`, this.publicKey.toBase58());
+    console.log('Is Connected: ', this.isConnected);
+    return { publicKey };
+  }
+
+  public async disconnect(): Promise<void> {
+    this.publicKey = null;
+    this.isConnected = false;
+  }
+
+  public async signAndSendTransaction<T extends Transaction | VersionedTransaction>(
+    transaction: T,
+    options?: SendOptions | undefined
+  ): Promise<{ signature: string }> {
+    console.debug('signAndSendTransaction');
+    const signedRawTx = (await this.signTransaction(transaction)).serialize();
+    console.debug('signAndSendTransaction: signedRawTx', signedRawTx);
+    const signature = await this.connection.sendRawTransaction(signedRawTx, options);
+    console.debug('signAndSendTransaction: signature', signature);
+    return { signature };
+  }
+
+  public async signTransaction<T extends Transaction | VersionedTransaction>(transaction: T): Promise<any> {
+    console.debug('signTransaction');
+    const instructions = this.getTransactionDetails(transaction);
+    const secretKeyBase58 = await this.callMethod(COMMON_REQUEST_MAP.wallet_getSigner, {
```

Comment:
This is bringing secrets into the browser's page context, isn't this dangerous? Allowing applications that are using the injection to potentially sniff it out? 

Normally, all required data for the transaction is passed into the wallet and signing happens in the background of the wallet. This ensures that private keys stay sandboxed in the wallet.

Reply:
Another issue that this brings up is what the user is prompted on the wallet. 
In this approach, they will agree to export their private key?

Injection requests should prompt the specific transaction

Reply:
Without this approach the impact will be huge:
- chainify must introcuce a new interface that will potentially impact other chains
-the solana package (in chainify) must be rewritten from scratch to use the latest solana web3js version

Reply:
In this case I don't think we can avoid it - key security is essential

> chainify must introcuce a new interface that will potentially impact other chains
This has been done before, see bitcoin's `signPSBT`, you can follow it through: 
https://github.com/liquality/wallet/blob/cd1982dc5bc9d38f0ed1e7e8e89f9667d4f58755/src/store/modules/app/requestPermission.js#L102
It's fairly separate and shouldn't affect other things. Just an extra interface on solana provider